pkg/block/indexheader/binary_reader.go

```diff
@@ -685,14 +685,10 @@ func (r BinaryReader) postingsOffset(name string, values ...string) ([]index.Ran
 		value := values[valueIndex]
 
 		i := sort.Search(len(e.offsets), func(i int) bool { return e.offsets[i].value >= value })
-		if i == len(e.offsets) {
-			// We're past the end.
+		if i == len(e.offsets) || e.offsets[i].value != value {
 			break
 		}
-		if i > 0 && e.offsets[i].value != value {
```

Comment:
I think we still need this (:

Reply:
why? Really I didn't get this logic.

Reply:
How we decrement the i then?

Reply:
Looks like This is still not clear, I think what would help is some extra unit test to verify

Reply:
Why do we need to decrement it? I mean that there is no value found then we need `break`.

Reply:
The whole idea behind this code is to find matching offset OR the largest offset that is before wanted value. Then we iterate one by one to find exact offset.

This is because binary index stores only 1/32 of offsets (to not keep all in memory). Problem is that our tests have only 3 postings most likely that's why we test nothing..

It's extremely concerning that our tests passed given that this version will literally skip 31/32 postings ((: